sopel/plugins/__init__.py

```diff
@@ -41,13 +43,22 @@
 
 from . import exceptions, handlers, rules  # noqa
 
+if TYPE_CHECKING:
+    from collections.abc import Iterable
 
-def _list_plugin_filenames(directory):
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _list_plugin_filenames(directory: str | os.PathLike) -> Iterable[tuple[str, str]]:
```

Comment:
This is interesting, because I distinctly remember having to report a bug in another library that refused to work on Python 3.8 because of `|` syntax in a type annotation, and suggested `Union` instead. Yet it seems to be fine here.

That said, stylistically, I believe we currently use `Union` everywhere else for this kind of annotation. I'm sure my search pattern was suboptimal, but searching the existing code for an annotation using `|` turned up nothing but false-positives from plugin command examples.